Error when read parquet file when there is special character in path string

**Describe the bug**
When there is a special character in path string, the parquet will read failed.

**To Reproduce**
Path:
```
hdfs://xxx/user/hive/wh/dw/xxx/dim_store_ware_snap/dt=20241102%7D/part-00011-be600396-f93c-4f90-aa02-302b75744f13.c000.snappy.parquet
```

Error logs:
```
 2024-12-03 14:51:32[Asia/Shanghai] - Caused by: java.lang.RuntimeException: poll record batch error: Execution error: native execution panics: Execution error: Execution error: output_with_sender[Shuffle] error: Execution error: output_with_sender[Filter] error: Execution error: output_with_sender[ParquetScan] error: Execution error: output_with_sender[ParquetScan]: output() returns error: Parquet error: External: Execution error: cannot get FSDataInputStream: $External("Java exception thrown at native-engine/datafusion-ext-commons/src/hadoop_fs.rs:57: java.io.FileNotFoundException: hdfs://xxx/user/hive/wh/dw/xxx/dim_store_ware_snap/dt=20241102%7D/part-00011-be600396-f93c-4f90-aa02-302b75744f13.c000.snappy.parquet: not found") 

```

**Expected behavior**
in version 4.0.0, the task will success. But in master branch, it will failed.

--- a/spark-extension-shims-spark3/src/test/scala/org/apache/spark/sql/blaze/BlazeQuerySuite.scala
+++ b/spark-extension-shims-spark3/src/test/scala/org/apache/spark/sql/blaze/BlazeQuerySuite.scala
@@ -1,0 +1,15 @@
+package org.apache.spark.sql.blaze
+
+import org.apache.spark.sql.Row
+
+class BlazeQuerySuite extends org.apache.spark.sql.QueryTest with BaseBlazeSQLSuite {
+
+  test("test partition path has url encoded character") {
+    withTable("t1") {
+      sql("create table t1 using parquet PARTITIONED BY (part) as select 1 as c1, 2 as c2, 'test test' as part")
+      val df = sql("select * from t1")
+      checkAnswer(df, Seq(Row(1, 2, "test test")))
+    }
+  }
+
+}

--- a/spark-extension/src/main/java/org/apache/spark/sql/blaze/JniBridge.java
+++ b/spark-extension/src/main/java/org/apache/spark/sql/blaze/JniBridge.java
@@ -17,6 +17,7 @@ package org.apache.spark.sql.blaze;
 
 import java.lang.management.BufferPoolMXBean;
 import java.lang.management.ManagementFactory;
+import java.net.URI;
 import java.util.List;
 import java.util.concurrent.ConcurrentHashMap;
 import org.apache.hadoop.fs.FileSystem;
@@ -85,11 +86,12 @@ public class JniBridge {
     }
 
     public static FSDataInputWrapper openFileAsDataInputWrapper(FileSystem fs, String path) throws Exception {
-        return FSDataInputWrapper$.MODULE$.wrap(fs.open(new Path(path)));
+        // the path is a URI string, so we need to convert it to a URI object, ref: org.apache.spark.paths.SparkPath.toPath
+        return FSDataInputWrapper$.MODULE$.wrap(fs.open(new Path(new URI(path))));
     }
 
     public static FSDataOutputWrapper createFileAsDataOutputWrapper(FileSystem fs, String path) throws Exception {
-        return FSDataOutputWrapper$.MODULE$.wrap(fs.create(new Path(path)));
+        return FSDataOutputWrapper$.MODULE$.wrap(fs.create(new Path(new URI(path))));
     }
 
     private static final List<BufferPoolMXBean> directMXBeans =

